pkg/runtime/core/scope_test.go

```diff
@@ -0,0 +1,39 @@
+package core_test
+
+import (
+	"testing"
+
+	"github.com/MontFerret/ferret/pkg/runtime/core"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestScope(t *testing.T) {
+	Convey("Should match", t, func() {
+		rs, cf := core.NewRootScope()
```

Comment:
Looks good.
Let's the final change.

RootScope - varA
Child1Scope - varB
Child2Scope - empty

Add a check that Child2Scope is able to get a variable from the root scope. (it will test functionality of scope traversing)

Reply:
You can just add as a 2nd Convey suit.

Reply:
I think the new `TestScopeTraversing` should cover this situation, let me know your thoughts :)